Documentation/Books/Manual/Programs/Arangorestore/ParallelRestore.md

```diff
@@ -1,2 +1,198 @@
+    } else {
+      pos = f.indexOf("_");
+      collName = f.slice(0, pos);
+    }
+  }
+  return {name: fullName, collName, size: fs.size(fullName)};
+});
+files = files.sort(function(a, b) { return b.size - a.size; });
+var dataFiles = [];
+for (let i = 0; i < files.length; ++i) {
+  if (files[i].name.slice(-10) === ".data.json") {
+    dataFiles.push(i);
+  }
+}
+
+// Produce the scripts, one for each coordinator:
+var scripts = [];
+for (let i = 0; i < coordinators.length; ++i) {
+  scripts.push([]);
+}
+
+var cnum = 0;
+for (let i = 0; i < dataFiles.length; ++i) {
+  var f = files[dataFiles[i]];
+  scripts[cnum].push(`${arangorestore} --collection ${f.collName} --input-directory ${dumpDir} --server.endpoint ${coordinators[cnum]} ` + otherArgs.join(" "));
+  cnum += 1;
+  if (cnum >= coordinators.length) {
+    cnum = 0;
+  }
+}
+
+for (let i = 0; i < coordinators.length; ++i) {
+  let f = "coordinator_" + i + ".sh";
+  print("Writing file", f, "...");
+  fs.writeFileSync(f, scripts[i].join("\n"));
+}
+EOF
+
+${ARANGOSH} --javascript.execute /tmp/parallelRestore$$.js -- "$@"
+rm /tmp/parallelRestore$$.js
+```
+
+To run this script, all coordinator endpoints of the cluster have to be
+provided. The example below is a template for a cluster with 3 coordinators:
+
+```
+./parallelRestore.js <dump-directory>
+  tcp://<ip-of-coordinator1>:<port of coordinator1>,
+  tcp://<ip-of-coordinator2>:<port of coordinator2>,
+  tcp://<ip-of-coordinator3>:<port of coordinator3>
+  --server.username <username>
+  --server.password <password>
+  --server.database <database_name>
+  --create-collection false
+```
+
+This time the option `--create-collection false` is used since we already
+created the document structure before. The above command will create 3 scripts,
+whereas 3 corresponds to the amount of listed coordinators. The resulting
+scripts are named _coordinator_number-of-coordinator.sh (e.g.
+coordinator_0.sh, coordinator_1.sh, coordinator_2.sh).
+
+### Execute parallel restore scripts
+
+The _coordinator_number-of-coordinator.sh scripts, that were created in the
+previous script, now have to be executed on each machine where a coordinator
+is running. This will start a parallel restore of the `dump`.
+
+### Reverse to the initial Replication Factor
+
+Once the _arangorestore_ process on every coordinator is completed, the
+replication factor has to be set to its initial value.
+
+Run the following command from exactly one coordinator (any coordinator can be
+used) - Please adjust the `replicationFactor` value accordingly.:
+
+```
+echo 'db._collections().filter(function(c) { return c.name()[0] !== "_"; })
+.forEach(function(c) { print("collection:", c.name(), "replicationFactor:",
+c.properties().replicationFactor); c.properties({ replicationFactor: 2 }); });'
+| arangosh
+  --server.endpoint <endpoint-of-a-coordinator>
+  --server.database <database-name>
+  --server.username <user-name>
+  --server.password <password>
+```
```

Comment:
Missing a verification step here? How to check if everything worked out?

Reply:
Yes I agree that a verification step should be added, unfortunately I do not have any log output currently. Omar told me that there is some kind of verification output.